.github/configs/sdkconfig.tls-psk

```diff
@@ -0,0 +1,5 @@
+# Workaround for https://github.com/espressif/esp-idf/issues/7631
```

Comment:
Can we avoid introducing another sdkconfig file? This would basically double the CI time, which is already long.

Reply:
My intent behind adding this file was to communicate the required SDK configuration fragments to users of MQTT PSK. If it makes explode CI times, I have no issue with removing it.

Is there another/better way of documenting the required sdk fragments?

Reply:
What about adding `CONFIG_ESP_TLS_PSK_VERIFICATION=y` with a comment to `.github/configs/sdkconfig.defaults`?

Reply:
Removed `sdkconfig.tls-psk` and replaced it with a hint in `sdkconfig.defaults`. Also removed TLS PSK support from the matrix build.

This results in TLS PSK no longer being regularly built in CI. I don't see how we could get these builds without the overhead from the matrix builds. But if this is a stepping stone towards getting this PR merged, let's do this step and increase CI build coverage for this feature in an upcoming one.